src/app/wallets/update-pending-invoices.ts

```diff
@@ -105,79 +105,76 @@ const updatePendingInvoice = async ({
+    // we're getting the invoice another time, now behind the lock, to avoid potential race condition
+    const invoiceToUpdate = await walletInvoicesRepo.findByPaymentHash(paymentHash)
+    if (invoiceToUpdate instanceof CouldNotFindError) {
+      pendingInvoiceLogger.error({ paymentHash }, "WalletInvoice doesn't exist")
+      return false
+    }
+    if (invoiceToUpdate instanceof Error) return invoiceToUpdate
+    if (invoiceToUpdate.paid) {
       pendingInvoiceLogger.info("invoice has already been processed")
       return true
     }
 
-    const lockService = LockService()
-    return lockService.lockPaymentHash({ paymentHash, logger, lock }, async () => {
-      const invoiceToUpdate = await walletInvoicesRepo.findByPaymentHash(
-        walletInvoice.paymentHash,
-      )
-      if (invoiceToUpdate instanceof CouldNotFindError) {
-        pendingInvoiceLogger.error(
-          { paymentHash: walletInvoice.paymentHash },
-          "WalletInvoice doesn't exist",
-        )
-        return false
-      }
-      if (invoiceToUpdate instanceof Error) return invoiceToUpdate
-      if (invoiceToUpdate.paid) {
-        pendingInvoiceLogger.info("invoice has already been processed")
-        return true
-      }
-
-      const updatedWalletInvoice = await walletInvoicesRepo.markAsPaid(
-        invoiceToUpdate.paymentHash,
-      )
-      if (updatedWalletInvoice instanceof Error) return updatedWalletInvoice
-
-      const usdPerSat = await getCurrentPrice()
-      if (usdPerSat instanceof Error) return usdPerSat
+    const invoicePaid = await walletInvoicesRepo.markAsPaid(paymentHash)
+    if (invoicePaid instanceof Error) return invoicePaid
 
-      const {
-        lnInvoice: { description },
-        roundedDownReceived,
-      } = lnInvoiceLookup
-      const fee = DepositFeeCalculator().lnDepositFee()
+    const usdPerSat = await getCurrentPrice()
+    if (usdPerSat instanceof Error) return usdPerSat
 
-      const usd = roundedDownReceived * usdPerSat
-      const usdFee = fee * usdPerSat
+    const {
+      lnInvoice: { description },
+      roundedDownReceived,
+    } = lnInvoiceLookup
+    const feeLightningLiquidity = DepositFeeCalculator().lnDepositFee()
 
-      const ledgerService = LedgerService()
-      const result = await ledgerService.addLnTxReceive({
-        walletId,
-        paymentHash,
-        description,
-        sats: roundedDownReceived,
-        fee,
-        usd,
-        usdFee,
-      })
-      if (result instanceof Error) return result
-
-      const notificationsService = NotificationsService(logger)
-      notificationsService.lnInvoicePaid({
-        paymentHash,
-        recipientWalletId: walletId,
-        amount: roundedDownReceived,
-        usdPerSat,
-      })
+    const usdDisplay = (roundedDownReceived * usdPerSat) as FiatAmount
+    const usdFeeLightningLiquidity = (feeLightningLiquidity * usdPerSat) as FiatAmount // TODO: toFiatFeeDisplay()
```

Comment:
we should standardize names. currently have two: bankFee (always in wallet currency) and networkFee (always sats), `usdFeeLightningLiquidity` is really confusing.